core/deployment/src/main/java/io/quarkus/deployment/steps/MainClassBuildStep.java

```diff
@@ -410,6 +412,14 @@ public MainClassBuildItem mainClassBuildStep(BuildProducer<GeneratedClassBuildIt
         return new MainClassBuildItem(mainClassName);
     }
 
+    private static String sanitizeMainClassName(ClassInfo classInfo) {
+        String className = classInfo.name().toString();
+        if (isKotlinClass(classInfo)) {
+            className += "Kt";
```

Comment:
I don't think this is safe at all.  There will be plenty of case (the majority in all likelihood) where `main()` will be part of a class definition and not a top level function as it is in the originating bug.

Reply:
Fixed